app/presentation/RoomItem/Touchable.tsx

```diff
@@ -28,255 +34,214 @@ interface ITouchableProps {
-		if (nativeEvent.oldState === State.ACTIVE) {
-			this._handleRelease(nativeEvent);
+	const onPress = () => {
+		if (rowState.value !== 0) {
+			close();
+			return;
+		}
+		const { onPress } = props;
+		if (onPress) {
+			onPress();
 		}
 	};
 
-	onLongPressHandlerStateChange = ({ nativeEvent }: any) => {
-		if (nativeEvent.state === State.ACTIVE) {
-			this.onLongPress();
+	const onLongPress = () => {
+		const { onLongPress } = props;
+		if (rowState.value !== 0) {
+			close();
+			return;
+		}
+
+		if (onLongPress) {
+			onLongPress();
 		}
 	};
 
-	_handleRelease = (nativeEvent: any) => {
-		const { translationX } = nativeEvent;
-		const { rowState } = this.state;
-		this._value += translationX;
+	const onLongPressHandlerStateChange = ({ nativeEvent }: any) => {
+		if (nativeEvent.state === State.ACTIVE) {
+			onLongPress();
+		}
+	};
 
+	const handleRelease = (event: any) => {
+		const { translationX } = event;
+		_value += translationX;
 		let toValue = 0;
-		if (rowState === 0) {
+		if (rowState.value === 0) {
 			// if no option is opened
 			if (translationX > 0 && translationX < LONG_SWIPE) {
-				// open leading option if he swipe right but not enough to trigger action
-				if (isRTL()) {
+				if (I18nManager.isRTL) {
 					toValue = 2 * ACTION_WIDTH;
 				} else {
 					toValue = ACTION_WIDTH;
 				}
-				this.setState({ rowState: -1 });
+				rowState.value = -1;
 			} else if (translationX >= LONG_SWIPE) {
 				toValue = 0;
-				if (isRTL()) {
-					this.hideChannel();
+				if (I18nManager.isRTL) {
+					hideChannel();
 				} else {
-					this.toggleRead();
+					toggleRead();
 				}
 			} else if (translationX < 0 && translationX > -LONG_SWIPE) {
 				// open trailing option if he swipe left
-				if (isRTL()) {
+				if (I18nManager.isRTL) {
 					toValue = -ACTION_WIDTH;
 				} else {
 					toValue = -2 * ACTION_WIDTH;
 				}
-				this.setState({ rowState: 1 });
+				rowState.value = 1;
 			} else if (translationX <= -LONG_SWIPE) {
 				toValue = 0;
-				this.setState({ rowState: 0 });
-				if (isRTL()) {
-					this.toggleRead();
+				rowState.value = 1;
+				if (I18nManager.isRTL) {
+					toggleRead();
 				} else {
-					this.hideChannel();
+					hideChannel();
 				}
 			} else {
 				toValue = 0;
 			}
-		}
-
-		if (rowState === -1) {
+		} else if (rowState.value === -1) {
 			// if left option is opened
-			if (this._value < SMALL_SWIPE) {
+			if (_value < SMALL_SWIPE) {
 				toValue = 0;
-				this.setState({ rowState: 0 });
-			} else if (this._value > LONG_SWIPE) {
+				rowState.value = 0;
+			} else if (_value > LONG_SWIPE) {
 				toValue = 0;
-				this.setState({ rowState: 0 });
-				if (isRTL()) {
-					this.hideChannel();
+				rowState.value = 0;
+				if (I18nManager.isRTL) {
+					hideChannel();
 				} else {
-					this.toggleRead();
+					toggleRead();
 				}
-			} else if (isRTL()) {
+			} else if (_value > ACTION_WIDTH * 2 && _value < LONG_SWIPE - FAV_SWIPE && I18nManager.isRTL) {
```

Comment:
I do not agree 100% with this, this animation is really short and you might accidentally activate `toggleFav` by just misswapping the `RoomItem`, so UX feels a bit uncomfortable. For now, lets just leave it out of this PR and we might evaluate this with our UX team later on. :)